collectd/collectd.go

```diff
@@ -107,7 +107,8 @@ func (s *Server) serve() {
 			continue
 		}
 		if n > 0 {
-			s.handleMessage(buffer[:n])
+			// Make packet processing in a separate goroutine so it prevents UDP packet loss
+			go s.handleMessage(buffer[:n])
```

Comment:
If we do this, we are using the buffer in the go-routine *and* this function. This may improve processing throughput, but to do it right, we must copy the bytes from `buffer` to a second buffer for use by the go-routine, right?